### DISCUSSION TITLE
Add support for Qwik as a UI Framework (@astrojs/qwik)

### DISCUSSION THREAD

**Discussion by virketrang** [withastro/roadmap#757] [Proposal]
### Body

# Summary

I would like to propose adding support for using Qwik as a UI Framework for Astro JS.

# Background & Motivation

Astro JS and Qwik offers the best performance in the JavaScript ecosystem. It would therefore be awesome to be able to combine the two. 

Futhermore, I think Qwiks resumability would play nicely with Astro's island architecture, allowing components using the client directive to only retrieve JavaScript when required. I also think this plays right into Astro's concept of shipping less JavaScript to the client.

# Goals

- Less unused JavaScript shipped to the client

# Example

You would simply add Qwik like any other UI Framework.
`
import { defineConfig } from 'astro/config';
import react from '@astrojs/qwik';

export default defineConfig({
  integrations: [qwik()],        
});
`

Then you would start using your Qwik components.

**Comment by ElianCodes** [comment]
Hey There! The Qwik devs are actually working on this! Super exciting! the package should be released soon under `@qwikdev/astro` and should also work with `npx astro add`. (The package is still private at this moment, but I'll update you if it's public) Closing the discussion in the meantime.
